dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateServiceCollectionExtensions.cs

```diff
@@ -4,108 +4,258 @@
+        ServiceLifetime lifetime = ServiceLifetime.Singleton)
+    {
+        Verify.NotNull(services);
+
+        services.Add(new ServiceDescriptor(typeof(WeaviateVectorStore), serviceKey, (sp, _) =>
+        {
+            var client = HttpClientProvider.GetHttpClient(clientProvider?.Invoke(sp), sp);
+            var options = GetStoreOptions(sp, optionsProvider ?? (static s => s.GetService<WeaviateVectorStoreOptions>()!));
+
+            return new WeaviateVectorStore(client, options);
+        }, lifetime));
+
+        services.Add(new ServiceDescriptor(typeof(VectorStore), serviceKey,
+            static (sp, key) => sp.GetRequiredKeyedService<WeaviateVectorStore>(key), lifetime));
+
+        return services;
+    }
+
+    /// <summary>
+    /// Registers a <see cref="WeaviateVectorStore"/> as <see cref="VectorStore"/>
+    /// with <see cref="WeaviateVectorStoreOptions"/> created with <paramref name="endpoint"/>, <paramref name="apiKey"/>.
+    /// </summary>
+    /// <inheritdoc cref="AddKeyedWeaviateVectorStore(IServiceCollection, object?, Func{IServiceProvider, HttpClient}, Func{IServiceProvider, WeaviateVectorStoreOptions}?, ServiceLifetime)"/>
+    [RequiresUnreferencedCode(DynamicCodeMessage)]
+    [RequiresDynamicCode(UnreferencedCodeMessage)]
     public static IServiceCollection AddWeaviateVectorStore(
         this IServiceCollection services,
-        HttpClient? httpClient = default,
+        Uri endpoint,
+        string? apiKey,
         WeaviateVectorStoreOptions? options = default,
-        string? serviceId = default)
+        ServiceLifetime lifetime = ServiceLifetime.Singleton)
+        => AddKeyedWeaviateVectorStore(services, serviceKey: null, endpoint, apiKey, options, lifetime);
+
+    /// <summary>
+    /// Registers a keyed <see cref="WeaviateVectorStore"/> as <see cref="VectorStore"/>
+    /// with <see cref="WeaviateVectorStoreOptions"/> created with <paramref name="endpoint"/>, <paramref name="apiKey"/>.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="WeaviateVectorStore"/> on.</param>
+    /// <param name="serviceKey">The key with which to associate the vector store.</param>
+    /// <param name="endpoint">The endpoint to connect to.</param>
+    /// <param name="apiKey">The API key to use.</param>
+    /// <param name="options">Options to further configure the <see cref="WeaviateVectorStore"/>.</param>
+    /// <param name="lifetime">The service lifetime for the store. Defaults to <see cref="ServiceLifetime.Singleton"/>.</param>
+    /// <returns>Service collection.</returns>
+    [RequiresUnreferencedCode(DynamicCodeMessage)]
+    [RequiresDynamicCode(UnreferencedCodeMessage)]
+    public static IServiceCollection AddKeyedWeaviateVectorStore(
+        this IServiceCollection services,
+        object? serviceKey,
+        Uri endpoint,
+        string? apiKey,
```

Comment:
This argument is nullable because for local testing, it can be null.
It's not optional, because I believe that it would be error prone for real-life scenarios (users missing the value in prod).